### PR TITLE
Fix spurious truncation warning

### LINKED ISSUE #1237: `-Wtruncation=1` warns on -$100, -$10000, etc.
This is an issue because `~$ff` is a valid 8-bit quantity (that can occur as a result of masking, etc.) that shouldn't be raising warnings. (`-Wtruncation=2` already exists for users that care about the sign bit.) This bug only occurs specifically on those boundary values.

### PR DESCRIPTION
**Pull request by Rangi42**
Fixes #1237

The line number changes obscure the test result, but compared to the previous RGBASM, it will only warn about `db ~$ff` and `dw ~$ffff` with `-Wtruncation=2`.

### CODE REVIEW

**Review by aaaaaa123456789 — APPROVED** [review]
LGTM

File: src/asm/rpn.cpp [hunk 1]
```diff
@@ -292,7 +292,7 @@ void rpn_CheckNBit(struct Expression const *expr, uint8_t n)
 	if (rpn_isKnown(expr)) {
 		int32_t val = expr->val;
 
-		if (val <= -(1 << n) || val >= 1 << n)
```
Comment (ISSOtm) [inline src/asm/rpn.cpp:295]: > `-Wtruncation=1` warns when an N-bit value's absolute value is 2**N or greater.

So this would be incorrect. -256 is truncated, after all: 256 is out of range.

**Review by ISSOtm — CHANGES_REQUESTED** [review]

File: src/asm/rpn.cpp (hunk 1 above)
Comment (Rangi42) [reply src/asm/rpn.cpp:295]: Despite the signed 32-bit internal storage in RGBASM, `~` conceptually just flips all the bits of a value. So if ~255, `%1111_1111`, is a valid 8-bit value, so should `~255` be.

Comment (Rangi42) [reply src/asm/rpn.cpp:295]: (It's true that this allows `-256` as well, but that's what `-Wtruncation=2` is for.)

Comment (ISSOtm) [reply src/asm/rpn.cpp:295]: No: level 1 was designed to accommodate negating an in-range value, level 2 being stricter and only allowing the Union of signed and unsigned range. Negating an out-of-range value is neither.

Comment (Rangi42) [reply src/asm/rpn.cpp:295]: Hmm. I agree with that logic regarding arithmetic negation. But regarding bitwise negation, I still think that if `N` is valid, `~N` must be also.

Maybe we can fit in a `-Wtruncation=0.5`?

Comment (aaaaaa123456789) [reply src/asm/rpn.cpp:295]: Level 1 was always intended to make 8-bit complements, whether bitwise or arithmetic (respectively, one's and two's), work without warnings. Not bringing up ~MAX back then, if it really wasn't brought up, would've been an oversight.

Comment (Rangi42) [reply src/asm/rpn.cpp:295]: #931 added `-Wtruncation` after multiple related issues and previous PR attempts (linked in its comments). In particular #212 complained that `~eightBitExpression` could result in wider values, for which `~$FF` is still the case. We were focused on the `~$80` case in that discussion, and on `-` arithmetic negation in #279, but `~$FF` wasn't addressed.

Comment (Rangi42) [reply src/asm/rpn.cpp:295]: How about:
```diff
 -Wtruncation=1 warns when
-an N-bit value's absolute value is 2**N or greater.
+an N-bit value is 2**N or greater, or less than -2**N.
 -Wtruncation=2 or just -Wtruncation also warns when
 an N-bit value is less than -2**(N-1),
 which will not fit in two's complement encoding.
```

Comment (ISSOtm) [reply src/asm/rpn.cpp:295]: That looks a lot like rule-patching to me, but eh, it's concise enough. Fine by me.

Comment (Rangi42) [reply src/asm/rpn.cpp:295]: Originally I could have said "`-Wtruncation=1` warns when an N-bit value is 2\*\*N or greater, or -2\*\*N or less"; the "absolute value" was for brevity (and should have clued me in, since there's always that off-by-one issue with signed negative vs positive ranges).

**Review by ISSOtm — APPROVED** [review]
